src/modules/paymentProcessor/PP_Streaming_v1.sol

```diff
@@ -690,7 +697,13 @@ contract PP_Streaming_v1 is Module_v1, IPP_Streaming_v1 {
             //Make sure to let paymentClient know that amount doesnt have to be stored anymore
             IERC20PaymentClientBase_v1(client).amountPaid(amount);
         } else {
-            unclaimableAmounts[client][paymentReceiver] += amount;
+            emit UnclaimableAmountAdded(
+                client, paymentReceiver, walletId, amount
+            );
+            //Adds the walletId to the array of unclaimable wallet ids
+            unclaimableWalletIds[client][paymentReceiver].push(walletId);
```

Comment:
Could there be a case where the `walletId` is already in that array? Would it be added twice then?

Reply:
Yes its possible, but if you look at the _claimPreviouslyUnclaimable() function the loop in line line 734 deletes the unclaimableAmount for that specific walletId. That means if the walletid is contained multiple times, then at the second occurence the unclaimableamount is 0 and therfore this interaction basically doesnt matter anymore

Reply:
fuu

Reply:
I just looked at it again and noticed that the unclaimableAmount Lookup is dependenat on it beign contained only once so i have to change it

Reply:
Here is the fix

Reply:
Good find

Reply:
Thanks, good stuff!